### ISSUE TITLE
Consolidate our install docs.

### ISSUE THREAD

**Issue by mattmoor** [knative/docs#2137]
tl;dr I would like to consolidate our install guides into a single source of truth.

### Background

knative.dev currently has over a dozen different guides for how to install parts of Knative. There are at least 8 guides for installing Knative that include cluster provisioning instructions. 
There are 4-5 guides for installing knative/serving with customized networking layers.

These guides are massively redundant, and for the most part only differ in how to provision a cluster with an upstream vendor.

### The Goal

I would like to see us consolidate a single source of truth for how to install a part or the whole of Knative.  Part of this is: Knative shouldn't be responsible for maintaining vendor-specific documentation.

### The Proposal

1. I would like to make `knative-with-any-k8s.md` the gold standard for how to install Knative, including documenting how to install pieces of Knative, and opting into distinct networking layers.

2. I would like to remove the networking-specific installation instructions (in favor of collapsable pieces inline in the above instructions, something like `<details>`)

<details><summary>To use Istio networking click here!</summary>
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
</details>

<details><summary>To use Gloo networking click here!</summary>
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum 
</details>

<details><summary>To use Kourier networking click here!</summary>
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum 
</details>

3. I would like to remove the vendor-specific installation instructions in favor of: links to vendor pages (they maintain) for how to consume Knative on their platform (What Openshift does here today is a great example), and collapsable "gotchas" inline in the above instructions.

<details><summary>Warning to Minikube users!</summary>
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
</details>

/assign

**Comment by mattmoor** [comment]
cc @samodell 

I am going to try to put together a PoC for `#1` now to see how viable a single source of truth could be.

**Comment by mattmoor** [comment]
I shared a little PoC in slack.

**Comment by mattmoor** [comment]
I published the latest from hackmd [here](https://github.com/mattmoor/docs/blob/rework-generic-install/docs/install/Knative-with-any-k8s.md)

**Comment by mattmoor** [comment]
@vaikas is going to take a crack at the eventing section in the hackmd later
